storey/targets.py

```diff
@@ -708,13 +723,18 @@ async def _worker(self):
                     if event is _termination_obj:  # handle outstanding batches and in flight requests on termination
                         for req in in_flight_reqs:
                             await self._handle_response(req)
-                        for shard_id in range(self._shard_count):
+                        for shard_id in range(self._shards):
                             if buffers[shard_id]:
                                 self._send_batch(buffers, in_flight_reqs, shard_id)
                         for req in in_flight_reqs:
                             await self._handle_response(req)
                         break
-                    shard_id = self._sharding_func(event) % self._shard_count
+                    shard_id = self._sharding_func(event)
```

Comment:
i prefer the kafka name you used (sharding_func_result) - since it's not (always) the shard_id at this point, and will be less confusing when we hash the shard_id down below